packages/jaeger-ui/src/components/DependencyGraph/DAG.js

```diff
@@ -41,17 +41,19 @@ export default class DAG extends React.Component {
     const nodes = [];
     const edges = [];
     serviceCalls.forEach(d => {
-      if (!nodeMap[d.parent]) {
-        nodes.push({ data: { id: d.parent } });
-        nodeMap[d.parent] = true;
+      if (d.parent.trim().length !== 0 && d.child.trim().length !== 0) {
```

Comment:
What are the semantics of `d.parent` and `d.child`? Could we not ensure elsewhere that they are not empty?

Could you add a unit test to demonstrate that data leads to an issue you're trying to fix?

Reply:
The way I was able to reproduce the issue is instrumenting a service with an space, (or empty string)

The context of this is `d.parent` and `d.child` are service names, I don't think there are other plkaces where we ensure that the service name is not empty, indeed I can see the empty service name in the search bar.

Is this considered an instrumentation error? Should we indicate it in some way?


I'll add a couple of tests.

Reply:
imo empty or whitespace service name should be considered an instrumentation error, and we could build a sanitizer to fix it during ingestion.

However, w.r.t. this code that looks things up in a map, an empty or whitespace string is still a string, so why is the code failing on that?

Reply:
I agree that we should build a sanitizer that fix this during ingestion, but also this view should not fail even if there are empty strings.

The error is not on this part, but on cytoscape, which doesn't allow empty strings nor white spaces as node IDs.